### DISCUSSION TITLE
Turn timeline physics on or off

### DISCUSSION THREAD

**Discussion by HolandonHdunto** [UPBGE/upbge#1710] [Ideas/Features for UPBGE 0.2.5+]
It would be nice if upbge had a way to enable or disable the physics of objects on the timeline, this is perfect to be used in hitbox where you determine the start and end of a collision has animation in real time.
    If this were done with the blocks of logic it would be confusing and unfeasible, in programming it would be a bit time consuming and you would probably have to revise the animation several times.

**Comment by BluePrintRandom** [comment]
```
import bge
cont = bge.logic.getCurrentController()
own = cont.owner

if own['FrameProperty']<own['Min'] or own['FrameProperty']>own['Max']:
    own.suspendPhysics()
else:
    own.restorePhysics()
```    
https://upbge.org/docs/latest/api/bge.types.KX_Object.html#/documentation/docs/latest/api/bge.types.KX_GameObject.html#bge.types.KX_GameObject.suspendPhysics

https://upbge.org/docs/latest/api/bge.types.KX_Object.html#/documentation/docs/latest/api/bge.types.KX_GameObject.html#bge.types.KX_GameObject.restorePhysics

**Comment by BluePrintRandom** [comment]
https://upbge.org/#/documentation/docs/latest/manual/manual/logic/actuators/types/edit_object.html

**Comment by HolandonHdunto** [comment]
I knew that I could suspend the physics of objects through the blocks of logic and programming, but for example I was thinking of making a hitbox system, in which a capsule of a target entity detects all the collisions of its attacker's abilities, in this case the idea would be to create custom skills where all you would have to do is create as a hitbox and put the skill name on each shard and simply animate, with each skill's information in a dictionary, and when the target entity's capsule detects a skill from an attacker, the Capsule would extract the information from each skill shard from the dictionary.
In a complex skill full of fragments it would be tricky, for example a combo with bear claw, has palm and bear claws, I would have to individually program the fragments, the start and end of the collision in the same animation several times.

Although the animation in bge is kind of buggy, and depending on the situation a single low poly character can devour almost all the fps in the game.
